src/things_mcp/fast_server.py

```
@@ -630,8 +614,15 @@ def run_things_mcp_server():
     else:
         logger.info("Things app is running and ready for operations")
         
+    # Log registered tool count — makes missing/broken tool registrations immediately visible
+    tool_count = len(mcp._tool_manager._tools)
+    logger.info(f"Registered {tool_count} tools")
```

Ruff (0.15.6)

[warning] Logging statement uses f-string

(G004)